lib/core/src/receive_swap.rs

```diff
@@ -32,6 +33,17 @@ pub(crate) struct ReceiveSwapHandler {
     liquid_chain_service: Arc<Mutex<dyn LiquidChainService>>,
 }
 
+#[async_trait]
+impl BlockListener for ReceiveSwapHandler {
+    async fn on_bitcoin_block(&self, _height: u32) {}
+
+    async fn on_liquid_block(&self, height: u32) {
+        if let Err(e) = self.claim_confirmed_lockups(height).await {
```

Comment:
Don't we want to claim also unconfirmed that are in the zero-conf range?

Reply:
The claim should have happened for `TransactionMempool` status already. There the tx is verified without the confirmation check. 

Shall we add this logic here also to only verify confirmation on non zero-conf swaps?

Reply:
I was thinking perhaps the code that should claim on TransactionMempool status might also not find the transaction in the mempool and skip the claim. But this is ok I think to focus on confirmed ones as it is a fallback after all.